scripts/release.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * modified from https://github.com/vitejs/vite/blob/main/scripts/release.ts
+ */
+import prompts from "prompts";
+import semver from "semver";
+import colors from "picocolors";
+import {
+  args,
+  getLatestTag,
+  getPackageInfo,
+  getVersionChoices,
+  isDryRun,
+  logRecentCommits,
+  run,
+  runIfNotDry,
+  step,
+  updateVersion,
+} from "./releaseUtils.js";
+
+let targetVersion: string | undefined;
+
+await logRecentCommits();
+
+const { currentVersion, pkgPath, pkgDir } = await getPackageInfo();
+
+if (!targetVersion) {
+  const { release }: { release: string } = await prompts({
+    type: "select",
+    name: "release",
+    message: "Select release type",
+    choices: getVersionChoices(currentVersion),
+  });
+
+  if (release === "custom") {
+    const res: { version: string } = await prompts({
+      type: "text",
+      name: "version",
+      message: "Input custom version",
+      initial: currentVersion,
+    });
+    targetVersion = res.version;
+  } else {
+    targetVersion = release;
+  }
+}
+
+if (!semver.valid(targetVersion)) {
+  console.error(`invalid target version: ${targetVersion}`);
+  process.exit(1);
+}
+
+const tag = `v${targetVersion}`;
+
+if (targetVersion.includes("rc") && !args.tag) {
+  args.tag = "rc";
+}
+if (targetVersion.includes("beta") && !args.tag) {
```

Comment:
idk if we care, but would this be better as an else if? no real diff I think unless we get a really messed up tag

Reply:
This was ripped from pagesjs which was ripped from Vite. I made some slight tweaks to it. The cleanness doesn't matter all that much.